service/model/project.go

```diff
@@ -0,0 +1,160 @@
+package model
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type ProjectRepository interface {
+	Create(ctx context.Context, p Project) (Project, error)
+	Read(ctx context.Context, id uuid.UUID) (Project, error)
+	ReadAll(ctx context.Context) ([]Project, error)
+	Update(ctx context.Context, p Project) (Project, error)
+	Delete(ctx context.Context, id uuid.UUID) error
+}
+
+var (
+	errProjectNameRequired = errors.New("project name is required")
+)
+
+type Project struct {
+	ID                 uuid.UUID
+	Name               string
+	SlackChannelID     string
+	ReleaseAlertConfig ReleaseAlertConfig
+	CreatedAt          time.Time
+	UpdatedAt          time.Time
+}
+
+type ProjectCreation struct {
+	Name           string
+	SlackChannelID string
+	RlsAlertCfg    ReleaseAlertConfig
+}
+
+type ProjectUpdate struct {
+	ProjectID         uuid.UUID
+	Name              *string
+	SlackChannelID    *string
+	RlsAlertCfgUpdate ReleaseAlertConfigUpdate
+}
+
+type ReleaseAlertConfig struct {
+	Message         string
+	ShowProjectName bool
+	ShowAppName     bool
+	ShowReleaseName bool
+	ShowChangelog   bool
+	ShowDeployments bool
+	ShowSourceCode  bool
+}
+
+type ReleaseAlertConfigUpdate struct {
+	Message         *string
+	ShowProjectName *bool
+	ShowAppName     *bool
+	ShowReleaseName *bool
+	ShowChangelog   *bool
+	ShowDeployments *bool
+	ShowSourceCode  *bool
+}
+
+func NewProject(name, slackChannelID string, cfg ReleaseAlertConfig) (Project, error) {
+	p := Project{
+		ID:                 uuid.New(),
+		Name:               name,
+		SlackChannelID:     slackChannelID,
+		ReleaseAlertConfig: cfg,
+		CreatedAt:          time.Now(),
```

Comment:
Define a variable `now` and use it for both fields, otherwise they might not be equal, which is weird considering no update happened.